okta-auth/public/index.html

```diff
@@ -0,0 +1,114 @@
+<div id="authenticated-user-content" hidden>
+    <h2>Authenticated with Firebase</h2>
+    <p id="user-info"></p>
+    <button onclick="firebase.auth().signOut();">Sign out</button>
+</div>
+<div id="signin-widget" hidden></div>
+<script>
+window.onload = async () => {
+    const oktaSignIn = new OktaSignIn({
+        baseUrl: OKTA_ORG_URL,
+        redirectUri: window.location.url,
+        authParams: {
+            display: 'page',
+        },
+        el: '#signin-widget',
+    });
+
+    if (oktaSignIn.hasTokensInUrl()) {
+        // The user has been redirected back to your app after signing in with
+        // Okta. Get the Okta access token from the response and use it to
+        // authenticate with Firebase.
+
+        // Get the access token from Okta.
+        const oktaTokenResponse = await oktaSignIn.authClient.token.parseFromUrl();
+        const accessToken = oktaTokenResponse.tokens.accessToken.value;
+        const oktaUserName = oktaTokenResponse.tokens.idToken.claims.name || '';
+        const oktaUserEmail = oktaTokenResponse.tokens.idToken.claims.email || '';
+
+        // Use the access token to call the firebaseCustomToken endpoint.
+        const firebaseTokenResponse = await fetch(CUSTOM_TOKEN_ENDPOINT, {
+            headers: {
+                'Authorization': `Bearer ${accessToken}`,
+            }
+        });
+        const firebaseToken = await firebaseTokenResponse.text();
+
+        // Use the Firebase custom token to authenticate with Firebase.
+        try {
+            await firebase.auth().signInWithCustomToken(firebaseToken);
+
+            // Now that the user is authenticated, you can use auth-enabled
+            // Firebase services. In this example, we update the Firebase
+            // user profile with information from Okta:
+            const user = firebase.auth().currentUser;
+            if (user.displayName !== oktaUserName) {
+                await user.updateProfile({displayName: oktaUserName});
+            }
+            if (user.email !== oktaUserEmail) {
+                await user.updateEmail(oktaUserEmail);
+            }
+        } catch (err) {
+            console.error('Error signing in with custom token.');
+        }
+    }
+
+    firebase.auth().onAuthStateChanged((user) => {
+        if (user) {
+            // User is signed in. Display some user profile information.
+            document.getElementById('user-info').innerHTML = `\
+                    <table>
+                        <tr>
+                            <th>Display name</th>
+                            <td>${user.displayName}</td>
+                        </tr>
+                        <tr>
+                            <th>Email address</th>
+                            <td>${user.email}</td>
+                        </tr>
+                        <tr>
+                            <th>Unique ID</th>
+                            <td>${user.uid}</td>
+                        </tr>
+                    </table>`;
```

Comment:
Could update this to
``` 
`Hi, ${user.displayName}! Your email address is
 ${user.email} and your UID is ${user.uid}.`;
```

Reply:
I like the table when code snippet length isn't an issue